src/tls.c

```diff
@@ -7130,6 +7133,98 @@ static int TLSX_KeyShare_GenEccKey(WOLFSSL *ssl, KeyShareEntry* kse)
     return ret;
 }
 
+#ifdef HAVE_LIBOQS
+/* Transform a group ID into an OQS Algorithm name as a string. */
+static const char* OQS_ID2name(int id)
+{
+    switch (id) {
+        case WOLFSSL_KYBER512:      return OQS_KEM_alg_kyber_512;
+        case WOLFSSL_KYBER768:      return OQS_KEM_alg_kyber_768;
+        case WOLFSSL_KYBER1024:     return OQS_KEM_alg_kyber_1024;
+        case WOLFSSL_NTRU_HPS2048509: return OQS_KEM_alg_ntru_hps2048509;
+        case WOLFSSL_NTRU_HPS2048677: return OQS_KEM_alg_ntru_hps2048677;
+        case WOLFSSL_NTRU_HPS4096821: return OQS_KEM_alg_ntru_hps4096821;
+        case WOLFSSL_NTRU_HRSS701:  return OQS_KEM_alg_ntru_hrss701;
+        case WOLFSSL_LIGHTSABER:    return OQS_KEM_alg_saber_lightsaber;
+        case WOLFSSL_SABER:         return OQS_KEM_alg_saber_saber;
+        case WOLFSSL_FIRESABER:     return OQS_KEM_alg_saber_firesaber;
+        case WOLFSSL_KYBER90S512:   return OQS_KEM_alg_kyber_512_90s;
+        case WOLFSSL_KYBER90S768:   return OQS_KEM_alg_kyber_768_90s;
+        case WOLFSSL_KYBER90S1024:  return OQS_KEM_alg_kyber_1024_90s;
+        default:                    return NULL;
```

Comment:
Would prefer break with `return NULL` at end of function.

Reply:
Got it.